buildpacks/metrics-agent/src/layers/install_agentmon.rs

```diff
@@ -0,0 +1,287 @@
+    > {
+        self.section
+            .say_with_details("Clearing cache", "invalid metadata");
+
+        Ok(libcnb::layer::MetadataMigration::RecreateLayer)
+    }
+}
+
+fn write_execd_script(agentmon: &Path, layer_path: &Path) -> Result<PathBuf, InstallAgentmonError> {
+    let log = layer_path.join("output.log");
+    let execd = layer_path.join("execd");
+    let daemon = layer_path.join("launch_daemon");
+    let run_loop = layer_path.join("agentmon_loop");
+
+    // Ensure log file exists
+    fs_err::write(&log, "").map_err(InstallAgentmonError::CouldNotWriteDestinationFile)?;
+
+    // agentmon_loop boots agentmon continuously
+    fs_err::copy(
+        additional_buildpack_binary_path!("agentmon_loop"),
+        &run_loop,
+    )
+    .map_err(InstallAgentmonError::CouldNotWriteDestinationFile)?;
+
+    // The `launch_daemon` schedules `agentmon_loop` to run in the background
+    fs_err::copy(additional_buildpack_binary_path!("launch_daemon"), &daemon)
+        .map_err(InstallAgentmonError::CouldNotWriteDestinationFile)?;
+
+    // The execd bash script will be run by CNB lifecycle, it runs the `launch_daemon`
+    fs_err::write(
+        &execd,
+        format!(
+            r#"#!/usr/bin/env bash
+
+                    {daemon} --log {log} --loop-path {run_loop} --agentmon {agentmon}
+                "#,
+            log = log.display(),
+            daemon = daemon.display(),
+            run_loop = run_loop.display(),
+            agentmon = agentmon.display(),
+        ),
+    )
+    .map_err(InstallAgentmonError::CouldNotCreateDestinationFile)?;
+
+    chmod_plus_x(&execd).map_err(InstallAgentmonError::PermissionError)?;
+
+    Ok(execd)
+}
+
+fn agentmon_download(dir: &Path) -> Result<PathBuf, InstallAgentmonError> {
+    download_to_dir(DOWNLOAD_URL, dir)?;
+
+    Ok(dir.join("agentmon"))
+}
+
+fn download_to_dir(url: impl AsRef<str>, destination: &Path) -> Result<(), InstallAgentmonError> {
+    let agentmon_tgz =
+        NamedTempFile::new().map_err(InstallAgentmonError::CouldNotCreateDestinationFile)?;
+
+    download(url, agentmon_tgz.path())?;
+
+    untar(agentmon_tgz.path(), destination)?;
+
+    chmod_plus_x(&destination.join("agentmon")).map_err(InstallAgentmonError::PermissionError)?;
+
+    Ok(())
+}
+
+pub(crate) fn untar(
+    path: impl AsRef<Path>,
+    destination: impl AsRef<Path>,
+) -> Result<(), InstallAgentmonError> {
+    let file = fs_err::File::open(path.as_ref()).map_err(InstallAgentmonError::CouldNotOpenFile)?;
+
+    Archive::new(GzDecoder::new(file))
+        .unpack(destination.as_ref())
+        .map_err(InstallAgentmonError::CouldNotUnpack)
+}
+
+/// Sets file permissions on the given path to 7xx (similar to `chmod +x <path>`)
+///
+/// i.e. chmod +x will ensure that the first digit
+/// of the file permission is 7 on unix so if you pass
+/// in 0o455 it would be mutated to 0o755
+pub fn chmod_plus_x(path: &Path) -> Result<(), std::io::Error> {
```

Comment:
This doesn't need to be `pub` (or even `pub(crate)`) at the moment. Using `pub` means Clippy's deadcode detection won't work. (If you plan to move this into a common library in the future then make it public at that point. For now nothing can use it from here anyway.)

```suggestion
fn chmod_plus_x(path: &Path) -> Result<(), std::io::Error> {
```